bin/makePsfLibrary.py

```diff
@@ -0,0 +1,175 @@
+        
+                    # convert the image to 64 bits, and turn it into a Psf
+                    # then warp it using the xytransform from above
+                    data = subImage.getArray().astype(numpy.float64)
+                    kernel = afwMath.FixedKernel(afwImage.ImageD(data))
+                    psf = measAlg.KernelPsf(kernel)
+                    warpedPsf = measAlg.WarpedPsf(psf, xytransform, "lanczos3");
+                    if len(outputHdus) == 0:
+                        hdu = pyfits.hdu.image.PrimaryHDU()
+                    else:
+                        hdu = pyfits.hdu.image.ImageHDU()
+                    hdu.data = warpedPsf.computeImage().getArray()
+                    hdu.header.append("GS_SCALE")
+                    hdu.header.set("GS_SCALE", .2)
+                    hdu.header.append("PSF_NO")
+                    hdu.header.set("PSF_NO",psfIndex)
+                    
+                    outputHdus.append(hdu)
+                    fout.write("%s %d %d\n"%(os.path.basename(output_file), psfCount, psfIndex))
+                    psfCount = psfCount + 1
+            psfIndex = psfIndex + 1
+        cat.close()
+        del hdu
+        del exp
+    outputHdus.writeto(output_file, clobber=True)
+    print psfCount, " psf images were cutout from the original image", len(outputHdus)
+
+#   This main program creates a library for each raft in the input_directory.  A raft is assumed
+#   to be comprised of 9 files named basename_Rnn_Smm.fits in the input directory
+#   The cutouts from each set of 9 files is placed in the output_dir in basename_Rnn.fits
+#   An index of all the libraries created in this way is created in basename.index
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("catalog", help="Name of the star catalog fed to OpSim")
+    parser.add_argument("input_dir", help="Name of the star grid produced by OpSim")
+    parser.add_argument("basename", help="Name of the star grid produced by OpSim")
+    parser.add_argument("output_dir", help="Name of the output directory")
+    args = parser.parse_args()
+    # List of available rafts for each focal plane
+    fout = open(args.output_dir + "/" + args.basename + ".index", "w")
+    for raft in rafts:
+        images = []
+        for file in os.listdir(args.input_dir):
+            if file.find(args.basename + "_" + raft + "_S") >= 0:
+                imagename = args.input_dir + "/" + file
+                if os.path.isfile(imagename):
+                    images.append(imagename)
+        catname = args.input_dir + "/" + args.catalog
+        output_file = args.output_dir + "/" + args.basename + "." + raft + ".psfs.fits"
+        #   If there are any images for this raft, cutout the Psfs and place in a mini-library
+        if len(images) > 0:
+            makeSubLibrary(images, args.input_dir + "/" + args.catalog, output_file, fout)
+        del images 
```

Comment:
Again, do you need to ``del``? If nothing else, ``images`` is just a list of strings -- unless you have truly enormous filenames, I doubt it's costing you much memory(!)